packages/xo-server/src/xo-mixins/remotes.js

```diff
@@ -84,6 +86,26 @@ export default class {
     return handler.test()
   }
 
+  async getAllRemotesInfo() {
+    const remotes = await this._remotes.get()
+
+    await asyncMap(remotes, async remote => {
+      try {
+        const handler = await this.getRemoteHandler(remote.id)
+        const info = await timeout.call(
+          handler.getInfo(),
+          this._remoteOptions.timeoutInfo
+        )
```

Comment:
What if the method returns info after the timeout?
Wouldn't it be interesting to save them in the cache?

Let's imagine a remote which returns info after `timeout + 1` delay, with your implementation we would never have any info, don't you think we could do better?

Reply:
I think it could be better, but I can't find a way.

We must use a timeout to avoid long calls so we throw an error in any case. So I can't figure out how to keep timeout, but still wait for a response from `getInfo`. Unless we use a kind of `setTimeout` to return an element after X seconds, and we take the `getInfo` out of `timeout.call()` to avoid throwing an error.

Reply:
```js
await timeout.call(
  handler.getInfo().then(info => {
    this._remotesInfo[remote.id] = info
  }),
  this._remoteOptions.timeoutInfo
)
```

Reply:
so simply.. :disappointed: 
Thanks !